File: Backend/VideoLibrary/Repository/Entities/Video.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Common.Model;

namespace OpenVisStreamer.VideoLibrary.Repository.Entities;

[Table("Videos")] 
public record Video
{
    [Key] 
    public Guid VideoId { get; set; } = Guid.NewGuid(); 

    [Required] 
    [StringLength(255)] 
    public string Title { get; set; }

    [Required] 
    public string Description { get; set; }

    [EnumDataType(typeof(VideoCategory))] 
    public VideoCategory Category { get; set; }= VideoCategory.Other;

    [Column(TypeName = "decimal(18, 2)")] 
    public decimal videoLength { get; set; }
    
    [Required] 
    public string VideoUri { get; set; }

    [Required] 
    public string ThumbnailUri { get; set; }

   
    [Required]
    public Guid uploadedByAccoutId { get; set; }
    
   
    public DateTime UploadDateTime { get; set; } = DateTime.Now;
    
}

File: Frontend/openvidstreamer-fe/src/Model/Video.ts
import {VideoCategory} from "./VideoUploadDTO.ts";

export type Video = {
    videoId: string;
    title: string;
    description: string;
    category: VideoCategory;
    videoLength: number;
    videoUri: string;
    thumbnailUri: string;
    totalLikes: number;
    totalDislikes: number;
    uploadDateTime: string; 
}

export function getCategoryName(categoryId: number): string {
    const categories = Object.entries(VideoCategory).map(([, value]) => value);
    return categories[categoryId] ?? "Unknown Category";
}
